### PR TITLE
feat: drop some capabilities to be never available

### PR DESCRIPTION
**Pull request by smira**
This PR makes sure that some capabilities (SYS_BOOT and SYS_MODULES) and
never be gained by any process running on Talos except for `machined`
itself.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

**Review by Unix4ever — APPROVED** [review]

File: internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go [hunk 1]
```diff
@@ -237,6 +238,49 @@ func SetRLimit(seq runtime.Sequence, data interface{}) (runtime.TaskExecutionFun
...
+		secbits |=
+			cap.SecbitNoCapAmbientRaise | cap.SecbitNoCapAmbientRaiseLocked
+
+		if err := secbits.Set(); err != nil {
+			return fmt.Errorf("error setting secbits: %w", err)
+		}
+
+		// Set PR_SET_NO_NEW_PRIVS to limit setuid and similar privilege raising techniques.
+		// See https://www.kernel.org/doc/html/v5.10/userspace-api/no_new_privs.html.
+		const PR_SET_NO_NEW_PRIVS = 38 //nolint:golint,revive,stylecheck
```
Comment (AlekSi) [inline internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go:255]: That's [`unix.PR_SET_NO_NEW_PRIVS`](https://pkg.go.dev/golang.org/x/sys/unix).

Comment (smira) [reply internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go:255]: thanks, not sure how I missed that

File: internal/app/machined/internal/install/install.go [hunk 2]
```diff
@@ -113,7 +114,14 @@ func RunInstallerContainer(disk, platform, ref string, configBytes []byte, reg c
 		oci.WithHostHostsFile,
 		oci.WithHostResolvconf,
 		oci.WithParentCgroupDevices,
-		oci.WithPrivileged,
+		oci.WithCapabilities(capability.AllGrantableCapabilities()),
```
Comment (smira) [inline internal/app/machined/internal/install/install.go:117]: I had to expand `Privileged` into explicit list, as in our case listing "all" capabilities for the reason that machined has different permitted and bounding capabilities.

**Review by andrewrynhard — DISMISSED** [review]

**Review by andrewrynhard — APPROVED** [review]

**Pull request merged**